arm-insights/2015-04-01/swagger/insightsClient.json

```diff
@@ -0,0 +1,1123 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get a page of tenant events",
+            "schema": {
+              "$ref": "#/definitions/EventDataCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/EventData"
+      }
+    },
+    "/{resourceUri}/providers/microsoft.insights/logDefinitions": {
+      "get": {
+        "tags": [
+          "LogDefinition"
+        ],
+        "operationId": "LogDefinition_Get",
+        "description": "The List Log Definitions operation lists the log definitions for the resource.",
+        "parameters": [
+          {
+            "name": "resourceUri",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource identifier of the target resource to get logs for."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2015-07-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get the last of log definitions",
+            "schema": {
+              "$ref": "#/definitions/LogDefinitionCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-odata": "#/definitions/LogDefinition"
+      }
+    },
+    "/providers/microsoft.insights/logs": {
+      "get": {
+        "tags": [
+          "Log"
+        ],
+        "operationId": "Log_Get",
+        "description": "The List Log operation lists the logs for the resource.",
+        "parameters": [
+          {
+            "name": "resourceUri",
+            "in": "body",
+            "required": false,
+"schema" : {},
+            "description" : "The call must return the logs for this resource id"
+          },
+          {
+            "name": "filterString",
+            "in": "body",
+            "required": false,
+"schema" : {},
+            "description" : "The filter to use in the query for logs"
+          },
+          {
+            "name": "definitions",
+            "in": "body",
+            "required": false,
+"schema" : {},
+            "description" : "The list of log definitions to use in the query"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
```

Comment:
I could not get the generated client to work correctly without adding:
```
{
   "$ref": "#/parameters/ApiVersionParameter"
}
```
Otherwise I would get an error that (?apiVersion=) is required for all service calls. Does this need to be added for each path entry?